### PR TITLE
Feat #545 [Reference Data] Refactor components and view models

### LINKED ISSUE #545: [Reference Data] Refactor components and view models
Currently, there is some duplicated code in Reference Data related classes. Create generic component and view model, and use them to refactor the existing ones.

### PR DESCRIPTION
**Pull request by joao4all**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/RHEAGROUP/COMET-WEB-Community-Edition/pulls) open
- [x] I have verified that I am following the COMET-WEB [code style guidelines](https://raw.githubusercontent.com/RHEAGROUP/COMET-WEB-Community-Edition/master/.github/CONTRIBUTING.md)
- [x] I have provided test coverage for my change (where applicable)

### Description
Closes #545 [Reference Data] Refactor components and view models

- Created a generic base component/viewmodel/row
- Existing classes related to reference data page are now refactored

### CODE REVIEW

File: COMETwebapp/Components/ReferenceData/SelectedReferenceDataItemBase.razor [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    License as published by the Free Software Foundation; either
+    version 3 of the License, or (at your option) any later version.
+    The CDP4-COMET WEB Community Edition is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+    Affero General Public License for more details.
+    You should have received a copy of the GNU Affero General Public License
+    along with this program. If not, see http://www.gnu.org/licenses/.
+------------------------------------------------------------------------------->
+@inherits DisposableComponent
```
Comment (jaimeatrhea) [inline COMETwebapp/Components/ReferenceData/SelectedReferenceDataItemBase.razor:17]: add a new line at the end please

File: COMETwebapp/Components/ReferenceData/SelectedReferenceDataItemBase.razor.cs [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+
+        /// <summary>
+        /// Method that is invoked when the edit/add thing form is being saved
+        /// </summary>
+        /// <returns>A <see cref="Task" /></returns>
+        protected virtual void OnEditThingSaving()
+        {
+        }
+    }
+}
```
Comment (jaimeatrhea) [inline COMETwebapp/Components/ReferenceData/SelectedReferenceDataItemBase.razor.cs:147]: New line at the end of the file

File: COMETwebapp/Pages/ReferenceData/ReferenceDataPage.razor [hunk 3]
```diff
@@ -20,25 +20,17 @@ Copyright (c) 2023-2024 RHEA System S.A.
 
 <DxToolbar ItemClick="@this.OnItemClick">
     <Items>
-        <DxToolbarItem Name="ParameterTypes" Text="Parameter Types" Tooltip="Parameter Types" />
-        <DxToolbarItem Name="MeasurementScales" Text="Measurement Scales" Tooltip="Measurement Scales" />
-        <DxToolbarItem Name="MeasurementUnits" Text="Measurement Units" Tooltip="Measurement Units" />
-        <DxToolbarItem Name="Categories" Text="Categories" Tooltip="Categories" />
+        @foreach (var mappedValue in this.mapOfComponentsAndNames)
+        {
+            <DxToolbarItem Name="@(mappedValue.Value)"
```
Comment (jaimeatrhea) [inline COMETwebapp/Pages/ReferenceData/ReferenceDataPage.razor:25]: If you are creating elements inside a loop. Use @key

**Review by antoineatrhea — APPROVED** [review]

File: COMETwebapp/Pages/ReferenceData/ReferenceDataPage.razor [hunk 4]
```diff
@@ -20,25 +20,17 @@ Copyright (c) 2023-2024 RHEA System S.A.
...
-        break;
-    case "MeasurementUnits":
-        <MeasurementUnitsTable />
-        break;
-    case "MeasurementScales":
-        <MeasurementScalesTable />
-        break;
-}
+    <DynamicComponent Type="@this.SelectedComponent" />
+}
```
Comment (jaimeatrhea) [inline COMETwebapp/Pages/ReferenceData/ReferenceDataPage.razor:36]: New line at the end of the file

**Review by jaimeatrhea — APPROVED** [review]
Small comments. No need to review it again

**Pull request merged**